src/couch_replicator_clustering.erl

```diff
@@ -42,94 +59,79 @@ start_link() ->
 
 
 % owner/2 function computes ownership for a {DbName, DocId} tuple
-% Returns {ok no_owner} in case no DocId is null. That case
-% would happen in the old replicator_manager if replication was
-% posted from _replicate endpoint and not via a document in 
-% *_replicator db.
-%
-% {error, unstable} value is returned if cluster membership has
-% been changing recently. Recency is a configuration parameter.
+% Returns `no_owner` in case no DocId is null, `unstable` if cluster
+% is considered to be unstable i.e. it has changed recently, or returns
+% node() which is considered to be the owner.
 %
 -spec owner(Dbname :: binary(), DocId :: binary() | null) ->
```

Comment:
Deciding on ownership based on `{dbname, docid}` means we don't make potentially a large number of external http requests to get filters (for js filters)  to compute replication ids. Currently we only look at document  info, then decide to skip it if it doesn't belong to the node. But if we disregard that corner case, having ownership decided based on #rep would be more elegant.

The other part involves trickery of what the return values mean. In this pr, ets table from replication manager is gone, so `nonode` should never be returned. But `no_owner` is a return value that means "there is no doc_id for this replication" i.e. this is `_replicate` replication whose ownership has already been determine on creation and after that it is not shuffled around.